internal/policy/policy.go

```diff
@@ -139,6 +141,76 @@ func (pa *PolicyAgent) Compile(files []*PolicyFile) error {
 	return nil
 }
 
+func (pa *PolicyAgent) initPolicyExcludeCache() map[string]bool {
+	cache := make(map[string]bool)
+	for _, policy := range pa.excludes.Policies {
+		cache["data."+policy] = true
+	}
+	return cache
+}
+
+func (pa *PolicyAgent) initGroupExcludeCache() map[string]bool {
+	cache := make(map[string]bool)
+	for _, g := range pa.excludes.PolicyGroups {
+		group := fmt.Sprint(g)
+		cache[group] = true
+	}
+	return cache
+}
+
+func isExcluded(s string, m map[string]bool) (bool, error) {
+	result, ok := m[s]
+	if !ok {
+		return false, fmt.Errorf("cache does not contain key: %q", s)
+	}
+	return result, nil
+}
+
+func (pa *PolicyAgent) compileModulesWithOpt(modules map[string]string, opts ast.CompileOpts) (*ast.Compiler, error) {
+
+	parsed := make(map[string]*ast.Module, len(modules))
+
+	policyExcludeCache := pa.initPolicyExcludeCache()
+	groupExcludeCache := pa.initGroupExcludeCache()
+
+module:
+	for f, module := range modules {
+		// Filter out tests
+		if strings.Contains(f, "test.rego") {
+			continue module
```

Comment:
It might be good to add log.debug with an info about skipped file, just in case - for future debugging